plasmapy/particles/particle_collections.py

```diff
@@ -0,0 +1,296 @@
+"""Collections of `~plasmapy.particles.particle_class.Particle` objects."""
+
+__all__ = ["ParticleList"]
+
+import astropy.units as u
+import collections
+import numpy as np
+
+from typing import *
+
+from plasmapy.particles.exceptions import *
+from plasmapy.particles.particle_class import (
+    CustomParticle,
+    DimensionlessParticle,
+    Particle,
+    ParticleLike,
+)
+
+
+class ParticleList(collections.UserList):
+    """
+    A `list` like collection of `Particle` and/or `CustomParticle` objects.
+
+    Parameters
+    ----------
+    particles : iterable
+        An iterable that provides a sequence of `ParticleLike` objects.
+        Objects that are not a `Particle` or `CustomParticle` instance
+        will be cast into a `Particle` instance.
+
+    Raises
+    ------
+    `~plasmapy.particles.exceptions.InvalidParticleError`
+        If an object supplied to `ParticleList` is not `ParticleLike`.
+
+    TypeError
+        If a `DimensionlessParticle` is provided.
+
+    Examples
+    --------
+    A `ParticleList` can be created by calling it with a `list`, `tuple`,
+    or other iterable that provides `~plasmapy.particles.ParticleLike`
```

Comment:
So I'm guessing the doc test glitches out on this reference, because it can't find ParticleLike. Do we have it in `particles/__init__.py`?

Reply:
Turns out we do. Note, however, that...

`/home/runner/work/PlasmaPy/PlasmaPy/docs/particles/index.rst.rst:64:autosummary: stub file not found 'plasmapy.particles.ParticleLike'. Check your autosummary_generate setting.`

double `.rst` extension for some reason?

Reply:
Something with https://github.com/namurphy/PlasmaPy/tree/particle-collection/docs/api_static (which I still don't quite get?) ? @rocco8773 do you see anything?

Reply:
It seems the current problem was related to me setting `ParticleLike.__name__` for some reason.  I just removed that line and I'm getting the tests to pass.  I'm going to raise a different issue about the docs for `ParticleLike`, since I started going beyond the scope of the current PR.

Reply:
That is pretty wack, indeed. Looks ready to merge, then!